### ISSUE TITLE
Add a generalized sensor postprocessor

### ISSUE THREAD

**Issue by harterj** [idaholab/moose#25590]
## Reason
In any multiphysics simulation, users may require sensor inquiry on properties of a system (temperature, pressure, radiation, etc).

## Design
The GeneralSensorPostprocessor functions as a base class, with inputs such as sensor noise mean and standard deviation, sensor delay, and provisions for a drift function. This class can be continually enhanced.

## Impact
This will add a new capability.

**Comment by Lin-INL** [comment]
Can the multiphysics simulation be treated as a mock-up unit (like in #23359), and digital engineer can extract, manipulate (adding noises, biases, delays, corrupted signals), and process information from it. Moreover, the control system designer can manipulate variables on the fly mimicking the real-time control scenarios.

**Comment by sallustius** [comment]
@Lin-INL, yes you should be able to do it (sorry I didn't see your comments). Feel free to reach out to me at my INL email if you want to chat more.
